### PR TITLE
Push gatekeeper with tag

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
Push the gatekeeper image with tag
Ref: https://issues.redhat.com/browse/ACM-9099

### CODE REVIEW

File: .github/workflows/build-push-stolostron.yaml [hunk 1]
```diff
@@ -24,5 +24,6 @@ jobs:
         REPOSITORY="quay.io/gatekeeper/gatekeeper" \
         PLATFORM="linux/amd64,linux/arm64,linux/arm/v8" \
         OUTPUT_TYPE=type=registry GENERATE_ATTESTATIONS=true \
+        VERSION=${{ github.ref_name }} \
```
Comment (dhaiducek) [inline .github/workflows/build-push-stolostron.yaml:27]: (See previous comment for details)
```suggestion
```

File: .github/workflows/build-push-stolostron.yaml
Comment (dhaiducek) [inline .github/workflows/build-push-stolostron.yaml:1]: Here's what I'm thinking here. Set this workflow to run on pushes to the branch:
```yaml
on:
  push:
    branches:
      - release-[0-9]+.[0-9]+
```

Add a step to tag the commit after the checkout step, with an increasing integer suffix:
```yaml
    - name: Tag commit
      run: |
        TAG="$(git tag -l "$(make print-version)-*" | sort --version-sort | tail -1)"
        if [[ -z ${TAG} ]]; then
          TAG=$(make print-version)-1
        else
          TAG="${TAG%-*}-$(( ${TAG#*-} + 1 ))"
        fi
        git tag -a ${TAG} -m ${TAG}
        git push ${TAG}
```

Then overwrite the tag in Quay with the version that matches upstream (we could also add another push in the workflow that references our tag like you have, but I'm not 100% sure that's necessary right now).

**Review by dhaiducek — COMMENTED** [review]
I've added some comments on how I was envisioning it might work--let me know what you think!

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ENABLE_PUBSUB ?= false
 AUDIT_CONNECTION ?= "audit"
 AUDIT_CHANNEL ?= "audit"
 
-VERSION := v3.15.1
+VERSION ?= v3.15.1
```
[the remark this answers is not in the document]
Comment (yiraeChristineKim) [reply Makefile:16]: Oh you're right!

[the remark this answers is not in the document]
Comment (yiraeChristineKim) [reply Makefile:16]: we don;t need this PR

[the remark this answers is not in the document]
Comment (yiraeChristineKim) [reply Makefile:16]: Thanks!!

File: .github/workflows/build-push-stolostron.yaml
Comment (yiraeChristineKim) [reply .github/workflows/build-push-stolostron.yaml:1]: great!!!

**Pull request closed without merging**